integration-tests/src/test/java/org/apache/druid/tests/security/ITBasicAuthConfigurationTest.java

```diff
@@ -77,6 +77,20 @@ public void test_druid99User_hasNodeAccess()
     checkNodeAccess(druid99);
   }
 
+  @Override
+  protected void setupHttpClientsAndUsers() throws Exception
+  {
+    super.setupHttpClientsAndUsers();
+
+    // Add a delay to allow propagation of credentials to all services
+    try {
+      Thread.sleep(5000);
```

Comment:
Do we know what changed here recently in auth land that this sleep is required? At any rate, I wonder if we could poll `/status` and/or issue a similar API call to verify things are setup instead of a static sleep.

Reply:
Yes, we recently introduced caching for basic auth in PR https://github.com/apache/druid/pull/15648 . That has sped up the authentication process triggering the flakiness in this test.

Only the coordinator exposes such APIs which can be used to fetch auth status, and indeed that API would return the expected response right away. The issue is the propagation of the creds from coordinator to other services (broker in this case) which may be delayed. Other services don't expose any API that would indicate the current set or version of auth creds that they have cached.